### PR TITLE
add some more context to error messages

### PR DESCRIPTION
**Pull request by Emilgardis**

### CODE REVIEW

**Comment by Emilgardis** [comment]
this is targeted towards #273.

@schrieveslaach could you install this fork/rev and try if it spits out a more helpful error message. The binary will be available in artifacts soon.

bors try --target x86_64-unknown-linux-gnu

**Comment by Emilgardis** [comment]
https://github.com/cross-rs/cross/suites/7158924821/artifacts/285015902

**Comment by schrieveslaach** [comment]
@Emilgardis, [here](https://gitlab.com/schrieveslaach/nextcloud-spgverein-app/-/jobs/2662014802) are the results.

**Comment by Emilgardis** [comment]
Awesome, thats a huge help, that file seems to be a symlink, so we're basically not handling that correctly

**Comment by Alexhuszagh** [comment]
> @Emilgardis, [here](https://gitlab.com/schrieveslaach/nextcloud-spgverein-app/-/jobs/2662014802) are the results.

That's interesting, normally you'd see all the symlinks being generated, which we use to keep the host and mount paths the same. I'll provide an example shortly.

**Comment by Alexhuszagh** [comment]
> Awesome, thats a huge help, that file seems to be a symlink, so we're basically not handling that correctly

Ah I definitely didn't handle them for copying the toolchains, it might have slipped through to ordinary files.

**Comment by Emilgardis** [comment]
> It should handle copying entire directories which contain symlinks fine, however.

The issue is this I believe. https://github.com/cross-rs/cross/blob/e583132ad33a3005ac3972a5467df16599791639/src/docker/remote.rs#L249

is_file == false for symlinks

**Comment by Alexhuszagh** [comment]
> 

That should be it. I'll submit a fix shortly. I did that intentionally for toolchains, but I also mistakenly applied that for projects and other mounted volumes.

File: src/docker/remote.rs [hunk 1]
```diff
@@ -130,7 +132,7 @@ fn copy_volume_files(
         .arg(src.to_utf8()?)
         .arg(format!("{container}:{}", dst.as_posix()?))
         .run_and_get_status(msg_info, false)
-        .map_err(Into::into)
+        .map(Into::into)
```
[the remark this answers is not in the document]
Comment (Alexhuszagh) [reply src/docker/remote.rs:135]: Apparently this doesn't need `map(Into::into)` anymore, so we can probably just remove it.

**Review by Alexhuszagh — CHANGES_REQUESTED** [review]
Once this is implemented, it's good to go:

> We should probably also add a warning if generating the symlinks fails, like if this subcommand fails. It should probably say something like `.wrap_err("when creating symlinks to provide consistent host/mount paths");`
> 
> https://github.com/cross-rs/cross/blob/e583132ad33a3005ac3972a5467df16599791639/src/docker/remote.rs#L955-L982

**Review by Alexhuszagh — APPROVED** [review]

**Pull request merged**